blog-EESSI-Cray-Slingshot11 (#551)
EESSI/docs: docs/blog/posts/2025/09/eessi-cray-slingshot11.md

In this blog post, we present the requirements for building OpenMPI 5.x with Slingshot 11 CXI support on Cray systems and its integration with EESSI using `host_injections`. This approach enables overriding EESSI’s default MPI library with an ABI-compatible, Slingshot-optimized version. The post concludes with test results validating this setup.

<!-- more -->

## The Challenge

EESSI provides a comprehensive software stack, but specialized interconnect support like Slingshot 11 CXI requires custom-built libraries that aren't yet available in the standard EESSI distribution. Our goal is to:

1. Build OpenMPI 5.x with native Slingshot 11 CXI support
2. Create ABI-compatible replacements for EESSI's MPI libraries
3. Support both x86_64 AMD CPU partitions and NVIDIA Grace CPU partitions with Hopper accelerators
4. Avoid dependency on system packages where possible

The main technical challenge is building the complete dependency chain on top of EESSI, as many of the required libraries for CXI support don't exist in the current EESSI stack.

## System Architecture

> Our target system consists of two distinct partitions:

Author: We are still in the pilot phase and will do this as soon as we move to production

mechanism of EESSI to inject custom-built OpenMPI libraries. This approach enables overriding EESSI’s default MPI library with an ABI-compatible, Slingshot-optimized version which should give us optimal performance.
<!-- more -->

## The Challenge

EESSI provides a comprehensive software stack, but specialized interconnect support like Slingshot-11 can sometimes require custom-built libraries that aren't yet available in the standard EESSI distribution. Our goal is to:

1. Build OpenMPI 5.x with native Slingshot-11 support
2. Create ABI-compatible replacements for EESSI's OpenMPI libraries
3. Place the libraries somewhere where EESSI automatically picks them up
4. Support both x86_64 AMD CPU partitions and NVIDIA Grace CPU partitions with Hopper accelerators

The main task is to build the required dependencies on top of EESSI, since many of the libraries needed for libfabric with CXI support are not yet available in the current EESSI stack.

### System Architecture

> Our target system is [Olivia](https://documentation.sigma2.no/olivia_pilot_period_docs/olivia_pilot_main.html) which is based on HPE Cray EX platforms for compute and accelerator nodes, and HPE Cray ClusterStor for global storage, all

Reviewer: I could imagine that the pilot page will be removed at some point. Maybe point to a different one?

---
author: [Richard]
date: 2025-09-10 
slug: EESSI-on-Cray-Slingshot
---

# MPI at Warp Speed: EESSI Meets Slingshot-11

High-performance computing environments are constantly evolving, and keeping pace with the latest interconnect technologies is crucial for maximizing application performance. The available interconnects, and the libraries

> that optimise their usage, is constantly shifting. EESSI can't constantly rebuild old software so how do we take advantage of new technological developments?

Reviewer: ```suggestion
High-performance computing environments are constantly evolving, and keeping pace with the latest interconnect technologies is crucial for maximising application performance. However, we cannot rebuild all the software in EESSI that depends on improvements to communication libraries. So how do we take advantage of new technological developments?
```

Specifically in this blog post we look at HPE/Cray supporting Slingshot-11 via the CXI [libfabric](https://ofiwg.github.io/libfabric/) provider. That's a lot of technical jargon, but it basically comes down to making a
software stack that can fully leverage the
capabilities of the interconnect hardware. Slingshot-11 promises to offer a significant advancement in HPC networking, offering improved bandwidth, lower latency, and better scalability for exascale computing workloads...so

> this should be worth the effort!

Reviewer: ```suggestion
Slingshot-11 promises to offer a significant advancement in HPC networking, offering improved bandwidth, lower latency, and better scalability for exascale computing workloads ... so
this should be worth the effort!
```

In this blog post, we present the requirements for building OpenMPI 5.x with Slingshot-11 support on HPE/Cray systems and its integration with EESSI using the [host_injections](../../../../site_specific_config/host_injections.md)

> mechanism of EESSI to inject the custom-built OpenMPI libraries. This approach enables overriding EESSI’s default MPI library with an ABI-compatible, Slingshot-optimized version which should give us optimal performance.

Reviewer: ```suggestion
mechanism of EESSI to inject custom-built OpenMPI libraries. This approach enables overriding EESSI’s default MPI library with an ABI-compatible, Slingshot-optimized version which should give us optimal performance.
```

256                   281.22
512                   559.04
1024                 1114.45
2048                 2081.25
4096                 4068.64
8192                 1852.11
16384               18564.47
32768               22647.40
65536               33108.03
131072              39553.95
262144              43140.01
524288              44853.40
1048576             45761.69
2097152             46228.10
4194304             46470.29

# OSU MPI-CUDA Latency Test v7.5
# Datatype: MPI_CHAR.
# Size       Avg Latency(us)
1                       2.76
2                       2.72
4                       2.90
8                       2.86
16                      2.85
32                      2.73
64                      2.60
128                     3.41
256                     4.17
512                     4.19
1024                    4.29
2048                    4.44
4096                    4.66
8192                    7.59
16384                   8.17
32768                   8.44
65536                   9.92
131072                 12.59
262144                 18.07
524288                 29.00
1048576                50.64
2097152                94.06
4194304               180.44
```

> ![OSU CUDA Bandwidth](osu_cuda_bandwidth.png)  ![OSU CUDA Latency](osu_cuda_latency.png) 

Author: Applied the suggested modifications
